File: Algorithms/454-4-Sum-II.swift
class Solution {
    func fourSumCount(_ A: [Int], _ B: [Int], _ C: [Int], _ D: [Int]) -> Int {
        var dict = [Int: Int]()
        for a in A {
            for b in B {
                let sum = a + b
                dict[sum, default: 0] += 1
            }
        }

        var result = 0
        for c in C {
            for d in D {
                let sum = c + d
                if let count = dict[-sum] {
                    result += count
                }
            }
        }

        return result
    }
}

File: Algorithms/454-4-Sum-II.py
class Solution(object):
    def fourSumCount(self, A, B, C, D):
        """
        :type A: List[int]
        :type B: List[int]
        :type C: List[int]
        :type D: List[int]
        :rtype: int
        """
        length = len(A)
        sum_m = dict()
        for i in range(length):
            for j in range(length):
                sum = A[i] + B[j]
                sum_m[sum] = sum_m.get(sum, 0) + 1

        ans = 0
        for i in range(length):
            for j in range(length):
                sum = -(C[i] + D[j])
                if sum in sum_m:
                    ans += sum_m[sum]

        return ans